<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="6.149" tests="16" failures="9">
  <testsuite name="Root Suite" timestamp="2024-06-03T22:41:22" tests="0" time="0.000" failures="0">
  </testsuite>
  <testsuite name="stock_trades_api_hard" timestamp="2024-06-03T22:41:22" tests="16" file="/home/yesayi/Desktop/2. NodeJS_ Stock Trades API project/test/index.spec.js" time="6.148" failures="9">
    <testcase name="stock_trades_api_hard should create a new trade of buy type" time="0.023" classname="should create a new trade of buy type">
    </testcase>
    <testcase name="stock_trades_api_hard should create a new trade of sell type" time="0.004" classname="should create a new trade of sell type">
    </testcase>
    <testcase name="stock_trades_api_hard should return 400 if the payload has an invalid type" time="0.002" classname="should return 400 if the payload has an invalid type">
    </testcase>
    <testcase name="stock_trades_api_hard should return 400 if the number of shares exceeds the limit" time="0.001" classname="should return 400 if the number of shares exceeds the limit">
    </testcase>
    <testcase name="stock_trades_api_hard should return 400 if the number of shares is below the limit" time="0.003" classname="should return 400 if the number of shares is below the limit">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 400 but got 201" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 400 but got 201
    at Context.<anonymous> (test/index.spec.js:100:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

      + expected - actual

      -201
      +400
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all the trades" time="0.020" classname="should fetch all the trades">
    </testcase>
    <testcase name="stock_trades_api_hard should fetch no trades if the type filter value does not exist" time="0.012" classname="should fetch no trades if the type filter value does not exist">
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all trades for a user" time="0.011" classname="should fetch all trades for a user">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500
    at Context.<anonymous> (test/index.spec.js:120:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

      + expected - actual

      -500
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch no trades if user filter value does not exist" time="0.010" classname="should fetch no trades if user filter value does not exist">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500
    at Context.<anonymous> (test/index.spec.js:127:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

      + expected - actual

      -500
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all buy trades for a user" time="0.008" classname="should fetch all buy trades for a user">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500
    at Context.<anonymous> (test/index.spec.js:134:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

      + expected - actual

      -500
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch all sell trades for a user" time="0.008" classname="should fetch all sell trades for a user">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500
    at Context.<anonymous> (test/index.spec.js:141:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

      + expected - actual

      -500
      +200
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should fetch a single trade" time="0.004" classname="should fetch a single trade">
    </testcase>
    <testcase name="stock_trades_api_hard should get 404 if the trade ID does not exist" time="0.002" classname="should get 404 if the trade ID does not exist">
      <failure message="expected &apos;&quot;ID not found&quot;&apos; to deeply equal &apos;ID not found&apos;" type="AssertionError"><![CDATA[AssertionError: expected '"ID not found"' to deeply equal 'ID not found'
    at Context.<anonymous> (test/index.spec.js:155:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

      + expected - actual

      -"ID not found"
      +ID not found
      ]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should get 405 for a put request to /trades/:id" time="2.002" classname="should get 405 for a put request to /trades/:id">
      <failure message="Timeout of 2000ms exceeded. For async tests and hooks, ensure &quot;done()&quot; is called; if returning a Promise, ensure it resolves. (/home/yesayi/Desktop/2. NodeJS_ Stock Trades API project/test/index.spec.js)" type="Error"><![CDATA[Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/yesayi/Desktop/2. NodeJS_ Stock Trades API project/test/index.spec.js)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should get 405 for a patch request to /trades/:id" time="2.002" classname="should get 405 for a patch request to /trades/:id">
      <failure message="Timeout of 2000ms exceeded. For async tests and hooks, ensure &quot;done()&quot; is called; if returning a Promise, ensure it resolves. (/home/yesayi/Desktop/2. NodeJS_ Stock Trades API project/test/index.spec.js)" type="Error"><![CDATA[Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/yesayi/Desktop/2. NodeJS_ Stock Trades API project/test/index.spec.js)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)]]></failure>
    </testcase>
    <testcase name="stock_trades_api_hard should get 405 for a delete request to /trades/:id" time="2.001" classname="should get 405 for a delete request to /trades/:id">
      <failure message="Timeout of 2000ms exceeded. For async tests and hooks, ensure &quot;done()&quot; is called; if returning a Promise, ensure it resolves. (/home/yesayi/Desktop/2. NodeJS_ Stock Trades API project/test/index.spec.js)" type="Error"><![CDATA[Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/yesayi/Desktop/2. NodeJS_ Stock Trades API project/test/index.spec.js)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)]]></failure>
    </testcase>
  </testsuite>
</testsuites>